yt_dlp/extractor/common.py

```diff
@@ -1452,8 +1452,16 @@ def extract_video_object(e):
             })
             extract_interaction_statistic(e)
 
-        for e in json_ld:
-            if '@context' in e:
+        def traverse_json_ld(json_ld, info, at_top_level=True):
+            for e in json_ld:
+                if at_top_level and '@context' not in e:
+                    continue
+                if at_top_level and all(k in ('@context', '@graph') for k in e):
+                    graph = e['@graph']
+                    if isinstance(graph, dict):
+                        graph = [graph]
+                    traverse_json_ld(graph, info, at_top_level=False)
+                    break
```

Comment:
```suggestion
        def traverse_json_ld(json_ld, at_top_level=True):
            for e in json_ld:
                if at_top_level and '@context' not in e:
                    continue
                if at_top_level and set(e.keys()) == {'@context', '@graph'}:
                    traverse_json_ld(variadic(e['@graph'], allowed_types=(dict,)), at_top_level=False)
                    break
```

1. the other similar functions are taking `info` from outer scope directly
2. `all(k in ('@context', '@graph') for k in e)` is just a complex way of saying `set(e) == {'@context', '@graph'}`

PS: `dict` should actually be added to `allowed_types` of `variadic` by default